frameworks/hdfs/tests/test_auth.py

```diff
@@ -87,28 +115,90 @@ def test_health_of_kerberized_hdfs():
     config.check_healthy(service_name=config.SERVICE_NAME)
 
 
+@pytest.fixture(scope="module", autouse=True)
 @pytest.mark.dcos_min_version('1.10')
 @sdk_utils.dcos_ee_only
-@pytest.mark.auth
-@pytest.mark.sanity
-def test_user_can_write_and_read(kerberos):
+def kerberized_hdfs_client(kerberos):
     try:
         client_app_def = config.get_kerberized_hdfs_client_app()
         client_app_def["secrets"]["hdfs_keytab"]["source"] = kerberos.get_keytab_path()
         client_app_def["env"]["REALM"] = kerberos.get_realm()
         client_app_def["env"]["KDC_ADDRESS"] = kerberos.get_kdc_address()
         sdk_marathon.install_app(client_app_def)
-        client_task_id = client_app_def["id"]
+        yield client_app_def["id"]
+
+    finally:
+        sdk_marathon.destroy_app(client_app_def["id"])
+
+
+@pytest.mark.dcos_min_version('1.10')
+@sdk_utils.dcos_ee_only
+@pytest.mark.auth
+@pytest.mark.sanity
+def test_user_can_auth_and_write_and_read(kerberized_hdfs_client):
+    sdk_auth.kinit(kerberized_hdfs_client, keytab=config.KEYTAB, principal=config.CLIENT_PRINCIPALS["hdfs"])
 
-        sdk_auth.kinit(client_task_id, keytab=config.KEYTAB, principal=config.GENERIC_HDFS_USER_PRINCIPAL)
+    write_cmd = "/bin/bash -c '{}'".format(config.hdfs_write_command(config.TEST_CONTENT_SMALL, config.TEST_FILE_1_NAME))
+    sdk_tasks.task_exec(kerberized_hdfs_client, write_cmd)
 
-        write_cmd = "/bin/bash -c '{}'".format(config.hdfs_write_command(config.TEST_FILE_1_NAME, config.TEST_CONTENT_SMALL))
-        sdk_tasks.task_exec(client_task_id, write_cmd)
+    read_cmd = "/bin/bash -c '{}'".format(config.hdfs_read_command(config.TEST_FILE_1_NAME))
+    _, stdout, _ = sdk_tasks.task_exec(kerberized_hdfs_client, read_cmd)
+    assert stdout == config.TEST_CONTENT_SMALL
 
-        read_cmd = "/bin/bash -c '{}'".format(config.hdfs_read_command(config.TEST_FILE_1_NAME))
-        _, stdout, _ = sdk_tasks.task_exec(client_task_id, read_cmd)
-        assert stdout == config.TEST_CONTENT_SMALL
 
-    finally:
-        sdk_marathon.destroy_app(client_task_id)
 
+@pytest.mark.dcos_min_version('1.10')
+@sdk_utils.dcos_ee_only
+@pytest.mark.auth
+@pytest.mark.sanity
+def test_users_have_appropriate_permissions(kerberized_hdfs_client):
+    # "hdfs" is a superuser
+    sdk_auth.kinit(kerberized_hdfs_client, keytab=config.KEYTAB, principal=config.CLIENT_PRINCIPALS["hdfs"])
+
+    for user, _ in config.CLIENT_PRINCIPALS.items():
```

Comment:
We could loop over `config.CLIENT_PRINCIPALS.keys()`.

Reply:
Note that we actually only use the `/users/alice` folder in the tests, so it is not required to create all of them.